api/src/main/java/ai/djl/util/Float16Utils.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.util;
+
+import java.nio.ByteBuffer;
+import java.nio.ShortBuffer;
+
+/** {@code Float16Utils} is a set of utilities for working with float16. */
+public final class Float16Utils {
+
+    private Float16Utils() {}
+
+    /**
+     * Converts a byte buffer of float16 values into a float32 array.
+     *
+     * @param buffer the buffer of float16 values as bytes.
+     * @return an array of float32 values.
+     */
+    public static float[] fromByteBuffer(ByteBuffer buffer) {
+        return fromShortBuffer(buffer.asShortBuffer());
+    }
+
+    /**
+     * Converts a short buffer of float16 values into a float32 array.
+     *
+     * @param buffer the buffer of float16 values as shorts.
+     * @return an array of float32 values.
+     */
+    public static float[] fromShortBuffer(ShortBuffer buffer) {
+        int index = 0;
+        float[] ret = new float[buffer.remaining()];
+        while (buffer.hasRemaining()) {
+            short value = buffer.get();
+            ret[index++] = halfToFloat(value);
+        }
+        return ret;
+    }
+
+    /**
+     * Converts an array of float32 values into a byte buffer of float16 values.
+     *
+     * @param floats an array of float32 values.
+     * @return a byte buffer with float16 values represented as shorts (2 bytes each).
+     */
+    public static ByteBuffer toByteBuffer(float[] floats) {
```

Comment:
You might want to take NDManager as parameter:
```
ByteBuffer buffer = manager.allocateDirect(floats.length * 2);
```
This can avoid one memory copy to native code.